tests/mtest/src/main/scala/tests/BaseSuite.scala

```diff
@@ -1,196 +1,59 @@
 package tests
 
-import scala.collection.mutable
 import scala.concurrent.Await
 import scala.concurrent.Future
 import scala.concurrent.duration.Duration
-import scala.language.experimental.macros
 import scala.meta.internal.metals.JdkSources
-import scala.meta.internal.metals.Testing
 import scala.meta.internal.mtags
 import scala.meta.internal.semver.SemVer
-import scala.meta.io.AbsolutePath
-import scala.reflect.ClassTag
 import scala.util.Properties
-import utest.TestSuite
-import utest.Tests
-import utest.asserts.Asserts
-import utest.framework.Formatter
-import utest.framework.TestCallTree
-import utest.framework.Tree
-import utest.ufansi.Attrs
-import utest.ufansi.Str
-
-/**
- * Test suite that replace utest DSL with FunSuite-style syntax from ScalaTest.
- */
-class BaseSuite extends TestSuite with TestOptionsConversions {
-  Testing.enable()
+import funsuite.Test
+import funsuite.TestOptions
+import funsuite.Location
+
+class BaseSuite extends funsuite.FunSuite with Assertions {
   def isJava8: Boolean =
     !Properties.isJavaAtLeast("9")
+
   def isScala211: Boolean =
     mtags.BuildInfo.scalaCompilerVersion.startsWith("2.11")
+
   def hasJdkSources: Boolean = JdkSources().isDefined
+
   def isWindows: Boolean =
     Properties.isWin
-  def isCI: Boolean =
-    "true".equalsIgnoreCase(System.getenv("CI"))
-  def beforeAll(): Unit = ()
-  def afterAll(): Unit = ()
-  def intercept[T: ClassTag](exprs: Unit): T = macro Asserts.interceptProxy[T]
```

Comment:
should we `override def isFlakyFailureOk = isCI`?

Reply:
BTW, we should be able to implement a basic `intercept` assertion that's not macro based
```scala
def intercept[T <: Throwable]body: => Unit)(implicit ev: ClassTag[T]): T = {
  try {
    body
    fail("expected exception but body evaluated successfully")
  }
  catch {
    case NonFatal(e) =>
      if (!ev.isAssignableFrom(e)) fail("expected ${ev.runtimeClass}, obtained ${e.getClass}")
  }
```

Reply:
I'm torn about this. What's the benefit of marking flaky tests and then ignoring them vs just ignoring them? Or a bit further: why don't we just remove those tests?

Reply:
as per `intercept`, I've added it to unblock this PR, although I think it would make sense in funsuite, so that it's properly tested